packages/editor/src/components/post-sync-status/index.js

```diff
@@ -1,35 +1,127 @@
 /**
  * WordPress dependencies
  */
-import { useSelect } from '@wordpress/data';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { PanelRow } from '@wordpress/components';
+import {
+	PanelRow,
+	Modal,
+	Button,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { ReusableBlocksRenameHint } from '@wordpress/block-editor';
 
 /**
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
 
 export default function PostSyncStatus() {
-	const { syncStatus, postType } = useSelect( ( select ) => {
+	const { syncStatus, postType, meta } = useSelect( ( select ) => {
 		const { getEditedPostAttribute } = select( editorStore );
 		return {
 			syncStatus: getEditedPostAttribute( 'wp_pattern_sync_status' ),
+			meta: getEditedPostAttribute( 'meta' ),
 			postType: getEditedPostAttribute( 'type' ),
 		};
-	}, [] );
+	} );
+
 	if ( postType !== 'wp_block' ) {
 		return null;
 	}
-
-	const isFullySynced = ! syncStatus;
+	// When the post is first created, the top level wp_pattern_sync_status is not set so get meta value instead.
+	const currentSyncStatus =
+		meta.wp_pattern_sync_status === 'unsynced' ? 'unsynced' : syncStatus;
```

Comment:
Is `meta` always an object? Just wondering if we need optional chaining here... feel free to ignore if it is always an object! 🙂

Reply:
In all my testing the entity always seems to have `meta` set as an object, even if it doesn't have any postmeta fields configured.

Reply:
Excellent. Thanks for confirming!

Reply:
actually, to be 100% correct after double checking, sometimes it is actually an empty array, in which case it still won't fail with an undefined error as never seems to be undefined so I don't think optional chaining is needed - but let me know if you have any concerns about that

Reply:
I updated to `meta?.wp_pattern_sync_status === 'unsynced' ? 'unsynced' : syncStatus`

Can't hurt, right?

Reply:
Thanks! Probably good to be on the safe side if a call to a selector doesn't necessarily guarantee that an object is returned. From a quick look up from `getEditedPostAttribute`, it winds up calling `getCurrentPostAttribute`, which internally calls `getCurrentPost`. If for some reason `getCurrentPost` returns its default empty object, then `getCurrentPostAttribute` could potentially return undefined around these lines: https://github.com/WordPress/gutenberg/blob/81ddf497dcc058f32d3ff08de48aebf2fe1bcf99/packages/editor/src/store/selectors.js#L280-L283

In practice, I doubt that'll ever be hit because (in theory) there'll always be a post object of some kind around, but this hardens the logic a little.

Cheers! 🍻

Reply:
Nice digging! We just might thank ourselves later 😄